lib/matplotlib/patches.py

```diff
@@ -1659,14 +1661,35 @@ def theta_stretch(theta, scale):
             theta = np.deg2rad(theta)
             x = np.cos(theta)
             y = np.sin(theta)
-            return np.rad2deg(np.arctan2(scale * y, x))
-        theta1 = theta_stretch(self.theta1, width / height)
-        theta2 = theta_stretch(self.theta2, width / height)
-
-        # Get width and height in pixels
-        width, height = self.get_transform().transform((width, height))
+            stheta = np.rad2deg(np.arctan2(scale * y, x))
+            # arctan2 has the range [-pi, pi], we expect [0, 2*pi]
+            return (stheta + 360) % 360
+
+        theta1 = self.theta1
+        theta2 = self.theta2
+
+        if (
+            # if we need to stretch the angles because we are distorted
+            width != height
+            # and we are not doing a full circle
+            and not (theta1 != theta2 and theta1 % 360 == theta2 % 360)
```

Comment:
probably needs a comment, IIRC you do this because theta1 and theta2 may not match anymore after stretching?

Reply:
can you instead first take everything mod 360 and then always stretch?

Reply:
I tried that and it did not work, as `0%360 == 360%360`.  I guess we could keep track of if `theta1 == theta2` up front and then add back 360 if they were?  I don't think we want to special case `theta1 == theta2` as input to mean "full circle" not "no arc".